src/cruise_control/scripts/cruise_control.py

```diff
@@ -5,34 +5,50 @@
 from sensor_msgs.msg import PointCloud
 from std_msgs.msg import Float64
 
-
 class CruiseControl:
 
     def __init__(self):
         self.steeringPub = rospy.Publisher("steering", Float64, queue_size = 10)
         self.brakingPub = rospy.Publisher("braking", Float64, queue_size = 10)
         self.throttlePub = rospy.Publisher("throttling", Float64, queue_size = 10)
+        self.speed = 0.0
 
     def listener(self):
         rospy.init_node("cruise_control", anonymous=True)
         rospy.Subscriber("lidar", PointCloud, self.handle_lidar_data)
         # Define Path data structure
         rospy.Subscriber("pathData", PathData, self.handle_path_data)
-        rospy.spin()
+        rospy.Subscriber("sensor/speed", Float64, self.handle_speed_data)
+
+        # Node is publisher and subscriber- cannot use spin; the publisher methods will never get called
+
+        # Probably should have same rate for this node and speedometer to make them more in sync
+        # Higher the rate the smoother the constant speed is, but will take more CPU power
+        rate = rospy.Rate(30)
+        while not rospy.is_shutdown():
+         self.publish_results()
```

Comment:
Because node is publisher and subscriber, the subscriber method needs to be called in a loop manually. This is why the publish_results method changes